List/recursive

```diff
@@ -13,18 +16,24 @@
             (alg < Neither = {=} | Both : { head : a, tail : b } >)
 
 in  let embed =
-            λ(fa : ../XNor/type  a (List a))
+            λ(a : Type)
+          → λ(fa : XNor a (List a))
           → merge
             { Both    = λ(q : { head : a, tail : List a }) → [ q.head ] # q.tail
             , Neither = λ(q : {}) → [] : List a
             }
             fa
 
 in    λ(a : Type)
```

Comment:
You should just pull this outside `let cata` rather than specifying the parameter in multiple places.